code/modules/virus2/items_devices.dm

```diff
@@ -1,74 +1,389 @@
+
+/obj/item/weapon/virusdish/attack_self(var/mob/user)
+	open = !open
+	update_icon()
+	to_chat(user,"<span class='notice'>You [open?"open":"close"] dish's lid.</span>")
+	if (open)
+		last_openner = user
+		if (contained_virus)
+			contained_virus.log += "<br />[timestamp()] Containment Dish openned by [key_name(user)]."
+		processing_objects.Add(src)
+	else
+		if (contained_virus)
+			contained_virus.log += "<br />[timestamp()] Containment Dish closed by [key_name(user)]."
+		processing_objects.Remove(src)
+	infection_attempt(user)
 
 /obj/item/weapon/virusdish/attackby(var/obj/item/weapon/W as obj,var/mob/living/carbon/user as mob)
 	..()
-	if(istype(W,/obj/item/weapon/hand_labeler) || istype(W,/obj/item/weapon/reagent_containers/syringe))
+	if(istype(W,/obj/item/weapon/hand_labeler))
 		return
-	if(user.a_intent == I_HURT)
+	if(user.a_intent == I_HURT && W.force)
 		visible_message("<span class='danger'>The virus dish is smashed to bits!</span>")
 		shatter(user)
 
+/obj/item/weapon/virusdish/is_open_container()
+	return open
+
+/obj/item/weapon/virusdish/afterattack(var/atom/A, var/mob/user, var/adjacency_flag, var/click_params)
+	. = ..()
+	if(.)
+		return
+
+	if (!adjacency_flag)
+		return
+
+	if (open)
+		if (istype(A,/obj/structure/reagent_dispensers))
+			var/obj/structure/reagent_dispensers/S = A
+			if(S.can_transfer(src, user))
+				var/tx_amount = transfer_sub(A, src, S.amount_per_transfer_from_this, user)
+				if (tx_amount > 0)
+					to_chat(user, "<span class='notice'>You fill \the [src] with [tx_amount] units of the contents of \the [A].</span>")
+					return tx_amount
+		if (istype(A,/obj/item/weapon/reagent_containers))
+			var/success = 0
+			var/obj/container = A
+			if (!container.is_open_container() && istype(container,/obj/item/weapon/reagent_containers) && !istype(container,/obj/item/weapon/reagent_containers/food/snacks))
+				return
+
+			if(A.is_open_container())
+				success = transfer_sub(src, A, 10, user, log_transfer = TRUE)
+
+			if (success > 0)
+				to_chat(user, "<span class='notice'>You transfer [success] units of the solution to \the [A].</span>")
+
+		if (istype(A,/obj/structure/toilet))
+			var/obj/structure/toilet/T = A
+			if (T.open)
+				empty(user,A)
+		if (istype(A,/obj/structure/urinal)||istype(A,/obj/structure/sink))
+			empty(user,A)
+
+/obj/item/weapon/virusdish/proc/empty(var/mob/user,var/atom/A)
+	if (user && A)
+		to_chat(user,"<span class='notice'>You empty \the [src]'s reagents into \the [A].</span>")
+	reagents.clear_reagents()
+
+/obj/item/weapon/virusdish/process()
+	if (!contained_virus || !(contained_virus.spread & SPREAD_AIRBORNE))
+		processing_objects.Remove(src)
+		return
+	if(world.time - last_cloud_time >= cloud_delay)
+		last_cloud_time = world.time
+		var/list/L = list()
+		L["[contained_virus.uniqueID]-[contained_virus.subID]"] = contained_virus
+		getFromPool(/obj/effect/effect/pathogen_cloud/core,get_turf(src), last_openner, virus_copylist(L), FALSE)
+		return 1
```

Comment:
This is suspicious. Returning 1 (should be `PROCESS_KILL` anyway) removes the dish from `processing_objects`, making the creation of a pathogen cloud only fire once (until the dish is reopened). Is this intended?

Reply:
Well it's not only unintended, but it doesn't actually seem to work this way. I've seen my dishes release several clouds.